frontend/app/src/store/blockchain/tokens.ts

```diff
@@ -102,6 +104,20 @@ export const useBlockchainTokensStore = defineStore('blockchain/tokens', () => {
     await fetchDetectedTokens();
   });
 
+  const { isTaskRunning } = useTasks();
+  const isDetecting = isTaskRunning(TaskType.FETCH_DETECTED_TOKENS);
+
+  const { fetchBlockchainBalances } = useBlockchainBalancesStore();
+
+  watch(isDetecting, async (curr, prev) => {
+    if (prev && !curr) {
+      await fetchBlockchainBalances({
```

Comment:
So this will trigger the refresh as soon as the task is done correct?

Reply:
yeah correct